Release pull request for branch 19.17

robot-clickhouse opened the pull request:
This PullRequest is part of ClickHouse release cycle. It's used by CI system only. Don't perform any changes with it.

Commit:
Auto version update to [19.17.2.1] [54428]

alesapin committed:
Forbidden to use column name more than once in insert query.

KochetovNicolai committed:
Merge pull request #7689 from ClickHouse/constants-and-remote

Fix remote constants.

(cherry picked from commit 2233660fe6707f2ace9f39396551871aeeb74c58)

alesapin committed:
Fix use of destroyed variable captured in lambda

alexey-milovidov committed:
Merge pull request #7506 from Enmk/odbc_bridge_nullable_fix

odbc table function now respects external_table_functions_use_nulls

(cherry picked from commit e14f395001586d017c41f8b78f8429407c7971b8)

alexey-milovidov committed:
Merge pull request #7726 from ClickHouse/fallback-for-unknown-cpus

Fallback for unknown CPUs when using libcpuid #5239

(cherry picked from commit f2bbfd995a20544571f1e37f47ddfbd62cc56e8b)

KochetovNicolai committed:
Merge pull request #7734 from ClickHouse/try-fix-integration-tests

Change dir for logs in test_storage_s3

CurtizJ committed:
fix order by with pk optimization

(cherry picked from commit f6b601cb9cee946436de935e66767ff53440f718)

KochetovNicolai committed:
Merge pull request #7756 from ClickHouse/fix-tests-with-globs

Fix tests with globs

(cherry picked from commit afa4a6e9228f905ea1111faedb1d794d9174f0fd)

Commit:
Auto version update to [19.17.2.4] [54428]

Commit:
Auto version update to [19.17.3.1] [54428]

alexey-milovidov committed:
Merge pull request #7760 from amosbird/clientfixexit

Fix multi query exit for clickhouse-client

(cherry picked from commit ed5e4cd4b5468456c28111e259d96c124e1e3b34)

alexey-milovidov committed:
Merge pull request #7743 from dmitriy-myz/fix-mysql-wire-auth

MySQL wire protocol: wrong source ip

(cherry picked from commit 698cb6c3eeab8147344a040840cd9b0bab8e5e51)

Commit:
Auto version update to [19.17.3.7] [54428]

Commit:
Auto version update to [19.17.4.1] [54428]

KochetovNicolai committed:
Merge pull request #7777 from ClickHouse/fix-array-intersect-overflow

Fix array intersect overflow

alexey-milovidov committed:
Merge pull request #7791 from DimasKovas/fix_insert_in_pod_array

Fix PODArray.insert

(cherry picked from commit 06b9466de0e7597855025f91bb3c633027fdb070)

alexey-milovidov committed:
Merge pull request #7789 from zhang2014/fix/ISSUES-7784

ISSUES-7784 disabled predicate optimize for WITH FILL expression

alexey-milovidov committed:
Merge pull request #7758 from 4ertus2/ast

Check if table has qualified column

vitlibar committed:
Merge pull request #7653 from vitlibar/skip-unknown-settings-with-warnings

Skip unknown settings with warnings.

akuzm committed:
Fix typo in Memory: reuse buffer if capacity allows. (#7797)

alexey-milovidov committed:
Merge pull request #7793 from DeifyTheGod/master

fix zero-access user creation

(cherry picked from commit 4d4de7da95e7973fbf084b7c9109210e36889ee5)

alexey-milovidov committed:
Merge pull request #7838 from filimonov/issue7383_update_libunwind

Disable DwarfFDECache in libunwind

alexey-milovidov committed:
Merge pull request #7870 from ClickHouse/fix_bytes_count_in_peekable_buffer

Fix float parsing in Values

alexey-milovidov committed:
Merge pull request #7873 from excitoon-favorites/fix-7558-7602

Fixed errors with space reservation introduced in #7558 and #7602

tavplubix committed:
Revert "Merge pull request #7653 from vitlibar/skip-unknown-settings-with-warnings"

This reverts commit 40cb97145d36d0e1281dfe629f8e38c87174442b.

tavplubix committed:
fix backward compatibility #7851

tavplubix committed:
Merge pull request #7762 from ClickHouse/possible_segfaults_in_table_functions

Fix Storage lifetime in table functions

Commit:
Auto version update to [19.17.4.11] [54428]

Commit:
Auto version update to [19.17.5.1] [54428]

alexey-milovidov committed:
Merge pull request #7816 from dingxiangfei2009/fix-copier-contention-master

Fix clickhouse-copier cleaning-tainting contention between concurrent workers

(cherry picked from commit eb7f48aaf61ead38360376a30c511fa98827e963)

alexey-milovidov committed:
Merge pull request #7913 from ClickHouse/prewhere_max_parallel_replicas

Fix prewhere with max_parallel_replicas.

(cherry picked from commit fed4d56df88a9180c16ff00b58c80ecced180d22)

alexey-milovidov committed:
Merge pull request #7911 from ClickHouse/prewhere_alias

Fix reading empty set with prewhere dependent on alias column.

(cherry picked from commit 18a70ba70e7754291151377c0f75d028ab46be42)

stavrolia committed:
Merge pull request #7908 from CurtizJ/enum_conversion

Fix enum conversion in native format for backward compatibility.

(cherry picked from commit 4d6214d33e58faa64f8737cab46e31fdf61337bc)

KochetovNicolai committed:
Merge pull request #7929 from ClickHouse/fix-union-with-diff-columns

Fix union with diff columns

(cherry picked from commit 4f794765113736bae2a2ad15ae154eb3927bec39)

filimonov committed:
attempt to fix kafka parsing performance regression https://github.com/ClickHouse/ClickHouse/issues/7261

filimonov committed:
Get rid of InputStream layer for Kafka, use InputStream directly. Reopening port during reset.

filimonov committed:
more progress - cleanup internal cycles

filimonov committed:
make virtual_columns local

filimonov committed:
get rid of cloneWithColumns in internal loop and some other trash

alesapin committed:
Merge pull request #7916 from azat/dict-avoid-RELOAD-on-CREATE-DATABASE

Do not reload *all* dictionaries on CREATE DATABASE

(cherry picked from commit af85eb8f80370a02e13b3866a65b62a43d34299c)

alexey-milovidov committed:
Merge pull request #7981 from ClickHouse/fix_number_of_streams_for_storages_with_globs

Limit number of streams for read from StorageFile and StorageHDFS

(cherry picked from commit e44dc0a6c8547d1fa8d46be12f555b018648668e)

alexey-milovidov committed:
Merge pull request #7983 from ClickHouse/fix_dictionary_lifetime

Fix bug with minimal dictionary lifetime equals zero

(cherry picked from commit 3355492f0272d25bd8c8e97f37f8cdc9b0b2572b)

alexey-milovidov committed:
Merge pull request #7979 from ClickHouse/fix_check_table_with_empty_pk

Fix bug in check table for tables without primary key

(cherry picked from commit c39bcf2b513410ff5fc40720ad0977f4581b94d5)

alexey-milovidov committed:
Update IInputFormat.cpp

alexey-milovidov committed:
Update ProtobufRowInputFormat.h

alexey-milovidov committed:
Update ValuesBlockInputFormat.cpp

alexey-milovidov committed:
Update ValuesBlockInputFormat.cpp

alexey-milovidov committed:
Update KafkaBlockInputStream.cpp

alexey-milovidov committed:
Merge pull request #7747 from hczhcz/patch-1113

Fix empty array handling in arraySplit

(cherry picked from commit edede563c9374cc246dbd7b80eae990d1ff78c5c)

alexey-milovidov committed:
Merge pull request #7755 from nikitamikhaylov/storage_set

Using the Set engine happen Exception: Method read is not supported by storage Set

(cherry picked from commit 812df3aef35d8e930abcb1c463abe717ec75a1a1)

filimonov committed:
Make the formats like RowBinaryWithNamesAndTypes work in Kafka & do not try to parse buffer when it at eof

KochetovNicolai committed:
Add forgotten rows number check to chunk constructor.

KochetovNicolai committed:
Fix num rows in IRowInputFormat

filimonov committed:
Fix Native format in Kafka

filimonov committed:
Fix for https://github.com/ClickHouse/ClickHouse/issues/8003 : old producers do not write timestamp, but column still should have enough rows

alesapin committed:
Merge pull request #8028 from ClickHouse/fix_libc_compatibility_of_odbc_bridge

Fix unintendent dependency from GLIBC 2.16 in clickhouse-odbc-bridge

(cherry picked from commit 2592e44eefa43a96a566b22ed647bb9a854030ba)

alesapin committed:
Merge pull request #8032 from ClickHouse/ddl_dictionaries_if_not_exists

Create dictionary if not exists

(cherry picked from commit cfa1b37fd66e87e2e5650daa182d115859d9a1ee)

alesapin committed:
Merge pull request #8029 from ClickHouse/fix_dict_partially_update_failover

Fix external dictionaries failover, when they have invalidate_query.

(cherry picked from commit a524e82a7f9024beb1b1089cd879792df2a40dc0)

alesapin committed:
Merge pull request #8047 from ClickHouse/fix_check_query_again

Fix bug in checkDataPart

(cherry picked from commit 6afafef86c1c7d65c28b3b25b7777808db13f1ed)

nikitamikhaylov committed:
Merge pull request #7786 from nikitamikhaylov/incorrect_count

Incorrect count() with condition query result for SummingMergeTree.

(cherry picked from commit de2d996063d23ba8ccd6f875314ff43f48a0474e)

filimonov committed:
Fix for timestamp & style

alesapin committed:
Merge pull request #8037 from vitlibar/system-reload-dictionary-causes-complete-reloading

SYSTEM RELOAD DICTIONARY reloads a dictionary completely

(cherry picked from commit de6cd490ef95c8a2bb9943d511f48e7d4f9a18ea)

nikitamikhaylov committed:
Merge pull request #7637 from nikitamikhaylov/limit_with_ties

Unexpected behaviour of LIMIT WITH TIES and LIMIT BY

(cherry picked from commit 5a45c7561dacf8521eb36c030af9aeba878361ec)

filimonov committed:
returning allowNext() to the proper position

Commit:
Auto version update to [19.17.5.18] [54428]

Commit:
Auto version update to [19.17.6.1] [54428]

alexey-milovidov committed:
Merge pull request #8071 from azat/Connection-sendCancel-SIGSEGV

Avoid null dereference after "Unknown packet X from server"

(cherry picked from commit bef3a0d2d2638b8f4b95fa954101aef1680ca5e6)

alesapin committed:
Merge pull request #8051 from ClickHouse/fix_collations_behaviour

Fix several bugs with collations.

(cherry picked from commit af7b8f9274ffc91cb290de41e458fac4ec18e84f)

alexey-milovidov committed:
Update KafkaBlockInputStream.cpp

alexey-milovidov committed:
Merge pull request #8084 from azat/read-temporary-tables-on-failure

Read temporary tables on failure to avoid interpreting that bits as query

(cherry picked from commit 9106b2854a43a8fd66b212f6bdbdf3a7db02960b)

alexey-milovidov committed:
Merge pull request #8054 from amosbird/glibc2.30

porting clock_gettime and clock_nanosleep

(cherry picked from commit 074aeb2bb811422558cd2ed65c1776cda61dd2cb)

alesapin committed:
Merge pull request #7810 from excitoon-favorites/deleteondestroy

Ignore redundant copies of parts after move and restart

(cherry picked from commit 650277afb0017c7fb83e05df266b7885a1bef3d9)

alesapin committed:
Merge pull request #8098 from ClickHouse/fix_dictionary_properties

Fix dictionary properties

(cherry picked from commit 397a3fc2e77fac38b5ea19746fa8c02b399dc788)

alesapin committed:
Merge pull request #8104 from excitoon-favorites/test_move_after_merge

FIxed behavior with ALTER MOVE ran immediately after merge finish moves superpart of specified

(cherry picked from commit 67f1ff1c8bcde3597aaa3790d5452b06cbb8cadb)

alesapin committed:
Push packages to lts

alesapin committed:
Merge branch '19.17' of github.com:ClickHouse/ClickHouse into 19.17

alexey-milovidov committed:
Merge pull request #8143 from Akazz/race_condition/process_ordinary_query

Resolved data race in DB::BlockStreamProfileInfo::calculateRowsBeforeLimit()

(cherry picked from commit 604a31bb66953f397eab3bd006e3bb68c8889543)

alexey-milovidov committed:
Merge pull request #8088 from dingxiangfei2009/ClusterCopier/refine-zxid-def

ClusterCopier: refine zxid definition
(cherry picked from commit d354628010480afc937eb146dde611b7546eef95)

alexey-milovidov committed:
Merge pull request #8176 from ClickHouse/fix_tmp_drop_relative_path

Fix loading of partially dropped table

(cherry picked from commit efb916c5569976de03ec7dbc875821f83b5fab7c)

alexey-milovidov committed:
Merge pull request #8185 from vitlibar/fix-drop-dictionary-if-exists

Fix DROP DICTIONARY IF EXISTS

(cherry picked from commit ad76a1bafb2b23680fabf81b30cef9eeb16a1007)

alexey-milovidov committed:
Merge pull request #7812 from infinivision/fix_part_mutation

fix part mutation

(cherry picked from commit 399995d60b9661828d6562d79c1984dbcfcf3eca)

alexey-milovidov committed:
Merge pull request #8204 from ClickHouse/nullable-rand

rand and nullable argument

(cherry picked from commit d0408e8265f78ff941a92c8af226adbc987818a3)

alexey-milovidov committed:
Merge pull request #7790 from bgranvea/simple-aggregate-function-fix

Fix "Sizes of columns doesn't match" error with SimpleAggregateFunction

(cherry picked from commit 8727f30ef83af7c1b6ffd2f099a62d3084e3463d)

alexey-milovidov committed:
Merge pull request #8164 from nicelulu/master

Fix trivial count query with ConvertingBlockInputStream.

(cherry picked from commit e0095840867001b1f9b3a9b8e73bb9303ccdb88d)

alexey-milovidov committed:
Merge pull request #8213 from ClickHouse/fix-exists-query

Fixed segfault in EXISTS query

(cherry picked from commit 3a265cd9f3e06df1669c6f7812ddd60efe1d2574)

alexey-milovidov committed:
Merge pull request #8197 from filimonov/fix_kafka_mv_subqueries

Init query context for Kafka to make subqueries work

(cherry picked from commit 38353d1253cbea23ead7e4facba178a448594757)

alexey-milovidov committed:
Merge pull request #8234 from ClickHouse/fix_insert_select_mysql

Fix insert select from mysql(...)

(cherry picked from commit dd58ddbc9a5587911b46ad44d0d52b7cabe8466d)

alexey-milovidov committed:
Applied a patch from Nicolae Vartolomei

(cherry picked from commit 4cf540b6218816e240c9a8c8582a78588672138a)

stavrolia committed:
Fix segfault in table function file while inserting into it (#8177)

* Fix segfault in table function file while inserting into it

(cherry picked from commit b47863a0c633795a33e931be0f20a6388c410d99)

alesapin committed:
Merge pull request #8127 from ClickHouse/restrictions_for_external_dictionaries_from_ddl

Add restrictions for several sources of external dictionaries

alexey-milovidov committed:
Merge pull request #8250 from ClickHouse/no_mutation_number_by_default

No mutation number by default

(cherry picked from commit c9887ccadc7b9c489af922edf5733016d6069960)

alesapin committed:
Merge pull request #8275 from ClickHouse/fix_range_dictionaries_ddl

DDL dictionaries Range fix

(cherry picked from commit 7b392d4e895444f00c588a37823c7481e44818c5)

KochetovNicolai committed:
Updated poco submodule (#8128)

alexey-milovidov committed:
Merge pull request #8241 from vitlibar/fix-allowed-client-hosts-checks

Fix checking if a client host is allowed.

(cherry picked from commit 0b70706f04195d60ce205dec6826d7e6b815a80e)

alexey-milovidov committed:
Fixed potential buffer overrun in decompress
* Fixed error in compression codecs.
* Added test
* One more check

(cherry picked from commits 5da506b, 056b2fe, 7833ee3)

vitlibar committed:
Merge pull request #8292 from vitlibar/update-simdjson

Fix memory leak in simdjson.

(cherry picked from commit 3ba66dd7042afbeb7fb24139e70828fd1734a384)

alexey-milovidov committed:
Merge pull request #8306 from ClickHouse/fix_rename_distributed

Fix rename distributed

(cherry picked from commit de01dd12b531b899249254613999a3b4cd53efdf)

alexey-milovidov committed:
Merge pull request #8082 from moon03432/master

fix bitmapAnd error when intersecting an aggregated bitmap and a scalar bitmap

(cherry picked from commit d52547dc756e75b3689684cf1137163dca8ef03f)

vitlibar committed:
Merge pull request #8342 from vitlibar/fix-checks-localhost-is-allowed

Fix checks if a client host is allowed in case it's the localhost.

(cherry picked from commit 297a5fd6278cffa4ef16e4fd891c2b541effec28)

Commit:
Merge pull request #8204 from ClickHouse/nullable-rand

rand and nullable argument

(fix after cherry pick from commit d0408e8)

KochetovNicolai committed:
Merge pull request #8351 from ClickHouse/fix-dict-with-clickhouse-source-and-subquery

Fix dict with clickhouse source and subquery

(cherry picked from commit 6541b2db9e228804f6346650b597c93fdff7fc13)

KochetovNicolai committed:
Merge pull request #8376 from ClickHouse/try-fix-tests-with-processors

Try fix tests with processors.

(cherry picked from commit 2e8cf5356f1031d9d1f144cde62ee32e007b6c61)

alexey-milovidov committed:
Merge pull request #8384 from ClickHouse/avoid-std-terminate-in-write-buffers

Avoid std::terminate in write buffers

(cherry picked from commit c949eb57f5fb32a1596cbdd0f834a11d83bd1df6)

alexey-milovidov committed:
Merge pull request #8383 from ClickHouse/fix-race-in-directory-monitor

Fix race in StorageDistributedDirectoryMonitor

(cherry picked from commit e2d2f02d5942488049c6b0f44e4dfdee8d46756b)

KochetovNicolai committed:
Remove context from formats (#8388)

* Remove Context from formats.

(cherry picked from commit 438ea5bc22efda6e5218fc5ad35af0360c1e7dd1)

KochetovNicolai committed:
Try fix build.

alesapin committed:
Merge pull request #8345 from azat/fix-flaky-test--Buffer_and_max_memory_usage

Fix buffer overcommit and 01019_Buffer_and_max_memory_usage flakiness

Commit:
Fix broken tests.

Commit:
Auto version update to [19.17.6.36] [54428]

Commit:
Auto version update to [19.17.7.1] [54428]

alexey-milovidov committed:
Merge pull request #8242 from achimbab/bugfix_bloomfilter

Fixed two bugs in the bloom_filter index.

weiqxu committed:
Fix build

weiqxu committed:
Fix it removes PID file by mistake In Daemon

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):

In Daemon mode, when the user start a new clickhouse, if there is old
runing process and the pid is same with the value in PID file, it will
not start. However it also remove the PID file of the old runing process by
mistake.

In this fix, it will not remove the pid file of old runing process when
it start failed.

alexey-milovidov committed:
Merge pull request #8508 from azat/create-table-as-dict-fix

Avoid SIGSEGV on CREATE TABLE .. AS dictionary

alexey-milovidov committed:
Merge pull request #8514 from azat/view-related-tests

VIEW tests fixes

alexey-milovidov committed:
Manuall cherry-pick of https://github.com/ClickHouse/ClickHouse/pull/8537

alexey-milovidov committed:
Merge pull request #8570 from infinivision/fix_dddl_hang

ensure network be up

alexey-milovidov committed:
Merge pull request #8314 from javisantana/fix/no_wait_for_no_active_replicas

check for active replicas when waiting for commands

alesapin committed:
Merge pull request #8588 from ClickHouse/fix_mutation_size_limit

Fix mutation size limit

alexey-milovidov committed:
Almost merge

alexey-milovidov committed:
Cherry pick by hand https://github.com/ClickHouse/ClickHouse/pull/8589

alexey-milovidov committed:
Merge pull request #8606 from ClickHouse/fix_defaults_mismatch_size

Fix mismatch column sizes when inserting defaults

alesapin committed:
Merge pull request #8614 from ClickHouse/fix_alter_codec

Fix alter codec bug

alexey-milovidov committed:
Mergeable code

alexey-milovidov committed:
Merge pull request #8613 from ClickHouse/azat-SimpleAggregateFunction-LowCardinality-merge-fix-v2

SimpleAggregateFunction(LowCardinality) fix (continuation)

alexey-milovidov committed:
Merge pull request #8602 from PerformanceVision/remove_partition

Remove outdated partition when a MergeTree database has been shutdown

alexey-milovidov committed:
Merge pull request #8625 from Vxider/fix-liveview-timestamp

fix liveview timestamp bug

alexey-milovidov committed:
Merge pull request #8653 from ClickHouse/fix_system_metadata_path

Fix metadata path for system database

alexey-milovidov committed:
Merge pull request #8674 from ClickHouse/fix-create-column-not-implemented-set

Fixed the issue "Method createColumn() is not implemented for data type Set"

alexey-milovidov committed:
Merge pull request #8663 from nikvas0/fix_lazy_db

Fix "DROP TABLE IF EXISTS" for lazy database

alexey-milovidov committed:
Merge pull request #8707 from ClickHouse/aku/log-hotfix

Avoid stale log files.

alexey-milovidov committed:
Cherry pick https://github.com/ClickHouse/ClickHouse/pull/8680

alexey-milovidov committed:
Merge pull request #8718 from Akazz/fixing_ternary_logic_for_batches

Fixed ternary logic (OR/AND/XOR) calculation in multiple batches

alexey-milovidov committed:
Merge pull request #8728 from Akazz/fixing_ternary_logic_for_batches

A followup to #8718: Simpler and faster code in logical functions

alesapin committed:
Fix build

alesapin committed:
Revert fix from https://github.com/ClickHouse/ClickHouse/pull/8589/files

alesapin committed:
Revert wrong revert

Commit:
First attempt to fix issue https://github.com/ClickHouse/ClickHouse/issues/7878

Commit:
Removing code used for debugging.

vzakaznikov committed:
Fixing styling issues.

alesapin committed:
Merge pull request #8763 from filimonov/kafka_fixes_backport_19_17

Kafka fixes backport 19 17

alesapin committed:
Remove redundant spaces

alesapin committed:
Merge pull request #8767 from ClickHouse/aku/null-format

Fix sending no data to client with Null format.

(cherry picked from commit 0d484be01ad99b6adb6b44b469f5aee5dd773e5d)

alesapin commented:
Fails of multiple volumes tests seem to be OK because they are known to be flaky and possible I have chery-picked something wrong. Performance failure is also OK because we don't have some performance optimizations in this branch. Smoke test for the split build is not important.

4ertus2 committed:
Merge pull request #8764 from 4ertus2/bugs

Fix toDecimalOrNull() with exception inside

(cherry picked from commit 289f16978f97afa1df3ed9db6f68a09da6e79313)

Commit:
Auto version update to [19.17.7.52] [54428]

Commit:
Auto version update to [19.17.8.1] [54428]

Commit:
Trying to fix issue with calculating default expressions. (Not working yet)

Commit:
Reworking fix for missing columns.

Commit:
Going back to NameOrDefault mode in ConvertingBlockInputStream
as the other approach caused test regressions. This time we just
erase the columns that need to be set to the default value or expression.

alexey-milovidov committed:
Merge pull request #8785 from ClickHouse/aku/empty-template-format

Do not crash if the row template file is empty.

(cherry picked from commit 48b3f33076ff6e3f38bb0ac0011408bd63d7bca2)

Commit:
Fixing NameOrDefault mode in ConvertingBlockInputStream.

KochetovNicolai committed:
Backport #8806

Try to remove some code.

Update PushingToViewsBlockOutputStream

Update PushingToViewsBlockOutputStream

Romve commented code.

fix 01019_materialized_view_select_extra_columns

alexey-milovidov committed:
Merge pull request #8791 from excitoon-favorites/fixdestructeddatapart

Fixed bug with `MergeTreeReadPool`

(cherry picked from commit de32863f73c793a0a7816e4a3c420b90557dc57f)

alexey-milovidov committed:
Merge pull request #8866 from achulkov2/issue-8865_unnamed_tuples_should_not_be_flattened

Issue 8865 unnamed tuples should not be flattened

filimonov committed:
Test for flushing by time and by block size (+fix), one extra test for virual columns

filimonov committed:
Test showing the problem with partial commit of polled batch + fix

KochetovNicolai committed:
Merge pull request #7796 from ClickHouse/processors-4.3

Processors and system.numbers

(cherry picked from commit 8ce6040306025ee438c2abeb6bcd90da3da3c894)

alexey-milovidov committed:
Merge pull request #8868 from amosbird/countoptfix

Fix `MakeSet` in Group By

(cherry picked from commit b567907431c2545797eb618808d8c4b013b26121)

filimonov committed:
Kafka proper rebalances processing, controlled assignment, faster tests

vitlibar committed:
Merge pull request #8907 from vitlibar/drop-temporary-table-removes-only-temporary

DROP TEMPORARY TABLE now drops only temporary table.

Commit:
Auto version update to [19.17.8.54] [54428]

Commit:
Auto version update to [19.17.9.1] [54428]

alexey-milovidov committed:
Merge pull request #8940 from ClickHouse/show-tables-not-like

Fixed ridiculous bug #8727

alexey-milovidov committed:
Merge pull request #8967 from CurtizJ/attach-if-not-exists

Fix query 'ATTACH TABLE IF NOT EXISTS'

4ertus2 committed:
Trying to fix early constant folding (#9010)

add setting to disable early constant folding

tavplubix committed:
fix integration tests

CurtizJ committed:
Merge pull request #8892 from PerformanceVision/select_constants

DB::Exception: Parameters are out of bound when we have a constant on the SELECT clause

tavplubix committed:
fix integration tests

tavplubix committed:
Merge pull request #8992 from filimonov/19.17_backfixes

19.17 backports

tavplubix committed:
Merge pull request #9026 from ClickHouse/fix_poco_ssl_timeouts

Fix poco secure sockets timeouts

tavplubix committed:
fix test

Commit:
Auto version update to [19.17.9.60] [54428]

Commit:
Auto version update to [19.17.10.1] [54428]

KochetovNicolai committed:
Fix typeo in ExternalModelsLoader.

KochetovNicolai committed:
Copy block info in ConvertingBlockInputStream.

(cherry picked from commit 350881a3da6a68dadcaa6c7ba976bc32150c1f3e)

tavplubix committed:
Merge pull request #9288 from ClickHouse/enable_ssl_in_functional_tests

Enable secure sockets in stateless tests

tavplubix committed:
Merge pull request #9302 from ClickHouse/fix_tests_with_secure_sockets

Try fix tests with secure sockets

alexey-milovidov committed:
Merge pull request #9262 from ClickHouse/fix_expressions_in_metadata

Fix expressions in metadata

KochetovNicolai committed:
Fix build.

KochetovNicolai committed:
Merge pull request #9362 from ClickHouse/add-max-exec-time-test

Added test with max_execution_time.

(cherry picked from commit b751c4e71f302ea2230da8ea9d1b9b265fcabc4e)

KochetovNicolai committed:
Added dists.xml

tavplubix committed:
fix config symlink

alesapin committed:
Merge pull request #9463 from ClickHouse/fix_mutations_with_insert_quorum

Trying to fix mutations with last quorum parts

alexey-milovidov committed:
Merge pull request #9507 from filimonov/kafka_write_suffix_exception

Proper processing of exception in write_suffix in Kafka

(cherry picked from commit 4629f78da789f0b80640d54ddb66a8ca8c064f14)

nikitamikhaylov committed:
Merge pull request #9513 from filimonov/kafka_exceptions_from_destructors

Kafka exceptions from destructors

(cherry picked from commit b3ad57a7235a06e53e467aee3d785d6f583b687b)

alexey-milovidov committed:
Merge pull request #9564 from ClickHouse/update-libunwind

Update libunwind for WSL

(cherry picked from commit 9b1795cec6c978b314f6af4ae54c0253500edbf7)

alexey-milovidov committed:
Merge pull request #9603 from ClickHouse/fix_cyclic_aliases

Fix simpliest case of cyclic aliases

(cherry picked from commit 0d7336c00e23e50216ebb691c6e3cc1d9c9d4a1e)

alexey-milovidov committed:
Merge pull request #9636 from azat/fix-sum-remote

Fix various sum*() functions via remote() usage (incorrect function name)

(cherry picked from commit 3ef1d40b9763c4f0a7b00de8eb3486241446da18)

alesapin committed:
Merge pull request #9734 from ClickHouse/fix_quoting_in_dictionary_ddl

Fix backquoting in dictionary ddl

(cherry picked from commit 72ba0323484e0511111c6ca16f4884336cd59c69)

alexey-milovidov committed:
Merge branch 'ParallelInputsProcessor-GlobalThreadPool-shutdown-fix' of https://github.com/azat/ClickHouse into azat-ParallelInputsProcessor-GlobalThreadPool-shutdown-fix

alexey-milovidov committed:
Merge pull request #9804 from azat/optimize_skip_unused_shards_type_mismatch

Fix SIGSEGV with optimize_skip_unused_shards when type cannot be converted

(cherry picked from commit 3550f401ece18706d1e04a4c9568a666ad8a3644)

alesapin committed:
Fix packager

alexey-milovidov committed:
Merge pull request #9833 from ClickHouse/fix-glibc-compatibility-3

Fix glibc compatibility

(cherry picked from commit 74ca9325ac4017d82e88f84e2b3bef19b4e56cd7)

Commit:
Fix compilation.

alesapin committed:
Fix runner script

alesapin committed:
Fix wrong protocol version check

alesapin committed:
Add missing configs for tests

alesapin committed:
Merge pull request #10117 from ClickHouse/fix_wrong_protocol_version_check

Fix wrong protocol version check

alesapin commented:
testing disabled until we fix directories structure.

abyss7 committed:
Apply renames to this branch

Commit:
Merge pull request #9447 from vitlibar/shard-clamps-settings-to-constraints

Shard now clamps the settings from the initiator.

(cherry picked from commit 5b4b40f86fe5959ed569483197afd5a6760b963c)

CurtizJ committed:
Merge pull request #10476 from ClickHouse/filimonov-docker-ca-certificates

add CA certificates to clickhouse-server docker image

(cherry picked from commit fa2385b26b0021fe65e69fdecc071c61a9a4729d)
(cherry picked from commit 60046c58186fbb50aaa118840e60d0ffb119089b)

alexey-milovidov committed:
Merge pull request #10491 from azat/dist-shutdown

Proper Distributed shutdown (fixes UAF, avoid waiting for sending all batches)

(cherry picked from commit c9334d3fde3972412e6dfe86cfa8bb3aa5c19d93)
(cherry picked from commit 15b6853fcc30201ff8558ddc2bf6b72bbab5f2de)

alexey-milovidov committed:
Merge pull request #10474 from hczhcz/patch-0422

Fix overflow in simpleLinearRegression

(cherry picked from commit 62e2f10959c54a88ca24a6de52abf2253cc9790d)
(cherry picked from commit 51d9d27dc5ca8456dc558ee64628493706b2fbf0)

alexey-milovidov committed:
Merge pull request #10513 from ClickHouse/fix-fractional-timezones-overflow

Fix overflow at beginning of unix epoch for fractional timezones

(cherry picked from commit 11bb3e3afa67390ed3f132a9ef388b87e2e793f5)
(cherry picked from commit 7a69363cc976083537c200e67ff3d32dd5ba344c)

alesapin committed:
Finalize mutations in background (#10526)


(cherry picked from commit 228dbf475e5a21091bfaa6b631c4df3d39d8c486)
(cherry picked from commit 8a6212b2ed509f5f53dc0fe62e93bf958b17bea4)

alexey-milovidov committed:
Merge pull request #10603 from ClickHouse/fix-mv-scalars

Fix mv scalars

(cherry picked from commit afd0938b99b60054177f5e99d6963dec23dd8432)

alexey-milovidov committed:
Fix flacky test #10603

4ertus2 committed:
add join_algorithm setting for rolling update

alesapin committed:
Fix build

Commit:
Auto version update to [19.17.10.1] [54428]

Commit:
Auto version update to [19.17.11.1] [54428]
